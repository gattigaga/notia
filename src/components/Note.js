import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Icon from "components/Icon";
import Tooltip from "components/Tooltip";
import CheckboxList from "components/CheckboxList";

const Container = styled.div`
  position: relative;
  background: white;
  padding: 16px;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.1);

  &:hover {
    box-shadow: 3px 3px 3px rgba(0, 0, 0, 0.2);
  }
`;

const Title = styled.h1`
  color: black;
  font-weight: bold;
  font-family: Roboto;
  font-size: 16px;
  margin-top: 0px;
  margin-bottom: 18px;
  cursor: default;
`;

const Text = styled.p`
  font-family: Roboto Slab;
  font-size: 14px;
  color: black;
  margin: 0px;
  line-height: 1.5em;
  cursor: default;
`;

const Note = ({ className, title, type, content, onChangeCheck }) => (
  <Container className={className}>
    <Title>{title}</Title>
    {type === "text" && <Text>{content}</Text>}
    {type === "list" && (
      <CheckboxList items={content} onChangeCheck={onChangeCheck} isStatic />
    )}
  </Container>
);

Note.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.oneOf(["text", "list"]),
  content: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string,
        isChecked: PropTypes.bool
      })
    )
  ]),
  onChangeCheck: PropTypes.func
};

export default Note;
