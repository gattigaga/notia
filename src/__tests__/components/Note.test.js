import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Note from "components/Note";

let wrapper, onClick;

beforeEach(() => {
  const onChangeCheck = jest.fn();
  const onClickClose = jest.fn();
  onClick = jest.fn();

  wrapper = shallow(
    <Note
      title="Learning"
      onChangeCheck={onChangeCheck}
      onClickClose={onClickClose}
      onClick={onClick}
    />
  );
});

describe("Note", () => {
  it("renders with type 'text'", () => {
    wrapper.setProps({
      type: "text",
      content: "I want to learn React"
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with type 'list'", () => {
    wrapper.setProps({
      type: "list",
      content: [
        {
          label: "Learn React",
          isChecked: true,
          isNew: false
        },
        {
          label: "Learn Redux",
          isChecked: true,
          isNew: false
        },
        {
          label: "Learn Redux Saga",
          isChecked: false,
          isNew: false
        }
      ]
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with custom class", () => {
    wrapper.setProps({
      className: "my-class"
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
