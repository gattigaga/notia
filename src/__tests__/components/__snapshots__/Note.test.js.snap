// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders with custom class 1`] = `
<styled.div
  className="my-class"
>
  <styled.h1>
    Learning
  </styled.h1>
</styled.div>
`;

exports[`Note renders with type 'list' 1`] = `
<styled.div>
  <styled.h1>
    Learning
  </styled.h1>
  <CheckboxList
    focusIndex={-1}
    isStatic={true}
    items={
      Array [
        Object {
          "isChecked": true,
          "isNew": false,
          "label": "Learn React",
        },
        Object {
          "isChecked": true,
          "isNew": false,
          "label": "Learn Redux",
        },
        Object {
          "isChecked": false,
          "isNew": false,
          "label": "Learn Redux Saga",
        },
      ]
    }
    onChangeCheck={[Function]}
  />
</styled.div>
`;

exports[`Note renders with type 'text' 1`] = `
<styled.div>
  <styled.h1>
    Learning
  </styled.h1>
  <styled.p>
    I want to learn React
  </styled.p>
</styled.div>
`;
